test/plugin/test_out_elasticsearch.rb

```diff
@@ -954,10 +1001,10 @@ def test_connection_failed_retry
       raise Faraday::ConnectionFailed, "Test message"
     end
 
-    driver.emit(sample_record)
-
-    assert_raise(Fluent::ElasticsearchOutput::ConnectionFailure) {
-      driver.run
+    assert_raise(Fluent::Plugin::ElasticsearchOutput::ConnectionFailure) {
```

Comment:
No, I don't.
Hard-coded magic number causes unstable test.
Instead, how about remove assert_raise in this case?
Because this test checks the counts of connection resets.

> So I think I understand now, the driver catches the exception and prints it, as seen here: https://github.com/fluent/fluentd/blob/master/lib/fluent/test/driver/base.rb#L141-L150. I don't understand why this behaviour is desirable, or how we can work around it.

Also I have no idea why fluentd v0.14 test driver choices current behavior.
I think that we should send feature request to https://github.com/fluent/fluentd if current behavior is undesirable.

> I guess the real question is why the driver is waiting until after_shutdown to flush, instead of flushing immediately.

Because v0.14 driver supports `try_write` (asynchronous write) feature.
If flushing immediately, it cannot support asynchronous write method testing.